draft-trammell-taps-post-sockets.md

```diff
@@ -818,12 +830,118 @@ func sayHelloWithAssociation(association Association) {
+  .Accept()-----------> [Carrier] -+----------> .Close()
+                         |     ^   |  close    [ Carrier  ]
+                         |     |   +- event -> [ (closed) ]
+                         |     |          
+              .Association()  .GetCarrier()
+                         |     |
+                         V     |
+                     [Association]   
+                           ^
+                           |
+              NewAssociation(local,remote)
+~~~~~~~
+{: #fig-carrier-lc title="Carrier and Association Life Cycle"}
+
+Access to more detailed information is possible through accessors on Carriers
+and Associations, following the relationships shown in {{fig-abstractions}}.
+Carriers allow their Transients to be accessed and enumerated, primarily for
+logging and debugging purposes; Associations likewise allow their Paths to be
+enumerated for access to cached path properties.
+
+Each Carrier has a .Send() method, by which Messages can be sent with given
+properties, and a .Ready() method, which supplies a callback for reading
+Messages from the remote side. .Send() is not available on Sinks, and .Ready()
+is not available on Sources. Carriers also provide .OnSent(), .OnAcked(), and
+.OnExpired() calls for binding default send event handlers to the Carrier, and
+.OnClosed() for handling passive close notifications.
+
+~~~~~~~
+                                  +---------[incoming]-----------+
+                                  |         [Message ]           V
+[outgoing] ---> .Send() --->  [Carrier] <---- .Ready() <---- [Receiver]
+[Message ]                        |
+                                  +--- .OnSent()
+                                  +--- .OnAcked()
+                                  +--- .OnExpired()
+                                  +--- .OnClosed()       
+~~~~~~~
+{: #fig-message-lc title="Sending and Receiving Messages and Events"}
+
+
+\[EDITOR'S NOTE (bht) on the text below: Parameters, as above, PolicyContext, and
+a Configuration are the same thing, as in
+https://github.com/mami-project/draft-trammell-post-sockets/issues/15 -- I'll
+use Configuration here, since that's where we're going, but won't modify this in
+the rest of the document]. 
+
+An application may have a global Configuation, as well as more specific
+Configurations to apply to the establishment of a given Association or Carrier.
+These Configurations are optional arguments to the Association and Carrier
+creation calls. Each Configuration is conceptually a key-value store, where
+information in more specific scopes overrides information in less specific
+scopes: application defaults can be overriden by specific Configurations bound
+to Carriers or Associations, all of which may be overriden by system or
+user-scoped configuration parameters. Configurations are also made directly
+available to protocol stack instances (PSIs, see {{sec-psi}}) for fine-grained
+control of implementation-specific configuration parameters.
```

Comment:
This paragraph seem to belong somewhere else

Reply:
hm, probably. It's left over from the first draft. Might need to be moved up to Configurations or just cut; i'll mark it with an ednote till we see where #23 lands.